soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -1785,6 +1791,20 @@ void FileChoose_Main(GameState* thisx) {
         gSaveContext.skyboxTime += 0x10;
     }
 
+    if (CVar_GetS32("gSkipLogoTitle", 0) && CVar_GetS32("gSaveFileID", 0) < 3) {
+        if (Save_Exist(CVar_GetS32("gSaveFileID", 0)) && FileChoose_IsSaveCompatible(Save_GetSaveMetaInfo(CVar_GetS32("gSaveFileID", 0)))) {
+            this->buttonIndex = CVar_GetS32("gSaveFileID", 0);
+            this->menuMode = FS_MENU_MODE_SELECT;
+            this->selectMode = SM_LOAD_GAME;
+        } else {
+            CVar_SetS32("gSaveFileID", 4);
```

Comment:
This will change the CVar permanently and result in essentially disabling Fast File Select until it is manually re-enabled. Maybe we should save the CVar value into a local variable and operate based on that? That way if the player puts the right OTR in place to make the save file compatible, Fast File Select won't have to be manually re-enabled first.

Reply:
Setting it back to a value that is valid seems okay for this situation IMO, but I can be convinced otherwise

Reply:
Right, so my main issue with this is that this will change the user's configuration file without their consent, It seems like this CVar is only used in this file, meaning that changing it should be easily avoidable. I have a similar situation with the `gMasterQuest` CVar, and I may refactor it at some point, but the reason I haven't yet is that the CVar is used across multiple files that don't necessarily have shared state I can use. This CVar is only used in this one file, so it should be easy to grab a copy of the current CVar value and work with that rather than actually changing the CVar globally (which is actually what the previous iteration of this feature did).

That way, if the user fixes the issue with their OTRs or whatever other setting that was preventing Fast File Select from working, they won't have to re-configure anything else to get it working again. Just feels like a better UX to me.

Reply:
so we need to account for a couple different situations here:
| only vanilla otr | only mq otr | both vanilla and mq otr | mq enabled | fast file select save type | behavior |
| - | - | - | - | - | - |
| :heavy_check_mark: | :x: | :x: | :x: | vanilla | load save |
| :heavy_check_mark: | :x: | :x: | :x: | mq | ? |
| :heavy_check_mark: | :x: | :x: | :x: | rando w/ only vanilla dungeons | load save |
| :heavy_check_mark: | :x: | :x: | :x: | rando w/ only mq dungeons | ? |
| :heavy_check_mark: | :x: | :x: | :x: | rando w/ both vanilla and mq dungeons | ? |
| :x: | :heavy_check_mark: | :x: | :heavy_check_mark: | vanilla | ? |
| :x: | :heavy_check_mark: | :x: | :heavy_check_mark: | mq | load save |
| :x: | :heavy_check_mark: | :x: | :heavy_check_mark: | rando w/ only vanilla dungeons | ? |
| :x: | :heavy_check_mark: | :x: | :heavy_check_mark: | rando w/ only mq dungeons | load save |
| :x: | :heavy_check_mark: | :x: | :heavy_check_mark: | rando w/ both vanilla and mq dungeons | ? |
| :x: | :x: | :heavy_check_mark: | :x: | vanilla | load save |
| :x: | :x: | :heavy_check_mark: | :x: | mq | ? |
| :x: | :x: | :heavy_check_mark: | :x: | rando w/ only vanilla dungeons | load save |
| :x: | :x: | :heavy_check_mark: | :x: | rando w/ only mq dungeons | load save |
| :x: | :x: | :heavy_check_mark: | :x: | rando w/ both vanilla and mq dungeons | load save |
| :x: | :x: | :heavy_check_mark: | :heavy_check_mark: | vanilla | ? |
| :x: | :x: | :heavy_check_mark: | :heavy_check_mark: | mq | load save |
| :x: | :x: | :heavy_check_mark: | :heavy_check_mark: | rando w/ only vanilla dungeons | load save |
| :x: | :x: | :heavy_check_mark: | :heavy_check_mark: | rando w/ only mq dungeons | load save |
| :x: | :x: | :heavy_check_mark: | :heavy_check_mark: | rando w/ both vanilla and mq dungeons | load save |

Reply:
Pretty much all of the question marks should result in going back to the file select screen, which should be correctly handled by this. There is one exception I can see right now, and that's the "both OTRs and MQ enabled" case.

The MQ checkbox only controls whether the save file gets created with the `isMasterQuest` value as true or false. As long as you have an MQ OTR you can play it even with the Checkbox disabled (basically the same way rando works, rando doesn't have to be enabled to play a previously created rando save file).

Reply:
tl;dr, the value of the `gMasterQuest` CVar is only relevant during save file creation, afterwards use a combo of `gSaveContext.isMasterQuest`, `ResourceMgr_GameHasMasterQuest()`, and `ResourceMgr_GameHasOriginal()`

Reply:
I’ll get this change made to not update the cvar 👍 

Also I think the weirdness with the gMasterQuest will go away once “the thing” is implemented

Reply:
Done here